### PR TITLE
Added support for building C/C++ libraries via bob

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: server/src/main/java/com/defold/extender/Extender.java [hunk 1]
```diff
@@ -976,14 +976,12 @@ public boolean accept(File pathname) {
         return outputFiles;
     }
 
-    private void buildExtension(File manifest, Map<String, Object> manifestContext) throws IOException, InterruptedException, ExtenderException {
+    private List<File> buildExtensionInternal(File manifest, Map<String, Object> manifestContext, List<File> srcDirs, File libraryOut) throws IOException, InterruptedException, ExtenderException {
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/Extender.java:979]: Mostly same as before, but can override the output library file

File: server/src/main/java/com/defold/extender/Extender.java [hunk 2]
```diff
@@ -1007,17 +1005,39 @@ private void buildExtension(File manifest, Map<String, Object> manifestContext)
...
+        srcDirs.add(new File(extDir, FOLDER_COMMON_SRC));
+        srcDirs.add(new File(extDir, FOLDER_ENGINE_SRC));
+        return buildExtensionInternal(manifest, manifestContext, srcDirs, null);
+    }
+
+    private List<File> buildLibrary(File manifest, Map<String, Object> manifestContext) throws IOException, InterruptedException, ExtenderException {
+        LOGGER.info("buildLibrary");
+        List<File> srcDirs = new ArrayList<>();
+        srcDirs.add(uploadDirectory);
+        String libName = String.format(platformConfig.writeLibPattern, manifestContext.get("extension_name"));
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/Extender.java:1039]: Making it easy to control the library name from bob

File: server/src/main/java/com/defold/extender/Extender.java [hunk 3]
```diff
@@ -2073,6 +2093,34 @@ private boolean shouldBuildEngine() {
     private boolean shouldBuildPlugins() {
         return shouldBuildArtifact("plugins");
     }
+    private boolean shouldBuildLibrary() {
+        return shouldBuildArtifact("library");
+    }
+
+    private List<File> buildLibraries() throws ExtenderException {
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/Extender.java:2100]: This is a shortened copy of buildEngine() without the linker step.

**Review by britzl — APPROVED** [review]
Looks good!

**Pull request merged**